docs/modules/ROOT/pages/documentation/payara-server/server-configuration/certificate-management.adoc

```diff
@@ -0,0 +1,490 @@
+
+[[command-options]]
+=== Command Options
+
+[cols="3,1,5,1,1",options="header"]
+|===
+|Option
+|Type
+|Description
+|Default
+|Mandatory
+
+|`--target`
+|String
+|The name of the instance to remove the certificate from.
+|server
+|no
+
+|`--listener`
+|String
+|The name of the HTTP or IIOP listener to remove the certificate from.
+|N/A
+|no
+
+|`--domain_name`
+|String
+|The name of the domain where the target instance exists.
+|domain1
+|no
+
+|`--domaindir`
+|String
+|The path to the directory containing the target domain.
+|${installDir}/glassfish/domains
+|no
+
+|`--alias` (primary)
+|String
+|The alias name of the certificate to remove.
+|N/A
+|yes
+
+|===
+
+[[example]]
+=== Example
+
+[source, shell]
+----
+asadmin> remove-from-keystore --domain_name production --target Instance1 --listener http-listener-2 mycert
+----
+
+[[remove-from-truststore]]
+== `remove-from-truststore`
+
+*Usage*::
+`asadmin> remove-from-truststore --target=instancename --listener=listenername alias`
+
+*Aim*::
+This command removes a certificate from the target instance or listener's trust store matching the provided alias.
+
+If the instance or listener is configured to use the default trust store, the command will instead synchronise
+the instance with the DAS (under the assumption the certificate has been removed from the default trust store of
+the DAS), since any certificates removed from the instance stores would be lost upon next synchronisation.
+
+[[command-options]]
+=== Command Options
+
+[cols="3,1,5,1,1",options="header"]
+|===
+|Option
+|Type
+|Description
+|Default
+|Mandatory
+
+|`--target`
+|String
+|The name of the instance to remove the certificate from.
+|server
+|no
+
+|`--listener`
+|String
+|The name of the HTTP or IIOP listener to remove the certificate from.
+|N/A
+|no
+
+|`--domain_name`
+|String
+|The name of the domain where the target instance exists.
+|domain1
+|no
+
+|`--domaindir`
+|String
+|The path to the directory containing the target domain.
+|${installDir}/glassfish/domains
+|no
+
+|`--alias` (primary)
+|String
+|The alias name of the certificate to remove.
+|N/A
+|yes
+
+|===
+
+[[example]]
+=== Example
+
+[source, shell]
+----
+asadmin> remove-from-truststore --domain_name production --target Instance1 --listener http-listener-2 mycert
+----
+
+
+[[remove-expired-certificates]]
+== `remove-expired-certificates`
+
+*Usage*::
+`asadmin> remove-expired-certificates --target=instancename --listener=listenername`
+
+*Aim*::
+This command removes all expired certificates from the target instance or listener's key and trust stores.
+
+If the instance or listener is configured to use the default trust store, the command will instead synchronise
+the instance with the DAS (under the assumption the certificate has been removed from the default trust store of
+the DAS), since any certificates removed from the instance stores would be lost upon next synchronisation.
```

Comment:
```suggestion
If the instance or listener is configured to use the default key or trust store, the command will remove the certificates from the default key and trust store of the DAS and then synchronise
the instance with the DAS. This is to preserve the certificate changes across instance restarts because any certificates removed directly from the instance stores would be added back upon next synchronisation.
```

Reply:
See comments above, your suggestion isn't what happens.